packages/RatPack.package/RPApplication.class/instance/initialize.st

```diff
@@ -5,8 +5,10 @@ initialize
 	
 	settings := RPSettings new.
 	
-	self onRequestDo: [:httpRequest | self processRequest: httpRequest];
+	self
+		initializeSettings;
 		initializeDynamicVariables;
 		initializeRoutes;
 		initializePragmaMiddleware;
-		initializeLogging.
\ No newline at end of file
+		initializeLogging;
+		onRequestDo: [:httpRequest | self processRequest: httpRequest].
```

Comment:
Well, the onRequestDo makes the HttpService we are subclassing from store that block. Now, when we receive a HttpRequest, it will call this block wil the request. This results in a race condition. This code will work perfectly fine if the initialization of our app finished before the first request comes in. It might lead to errors if we directly receive a request before initialization is finished. By reordering, this race condition is resolved.